### ISSUE TITLE
Download preexisting SM torrent if it is a dupe

### ISSUE THREAD

**Issue by damonjavert** [damonjavert/jps2sm#17]
2020-12-28T20:27:06Z

So users can cross seed easily.

tldr

Download the exisiting SM torrent upon hitting a dupe so user can seed it.

When jps2sm finds a dupe it simply errors:

```
        SMerrorTorrent = re.findall('red; text-align: center;">(.*)</p>', SMres.text)
        if SMerrorTorrent:
            dupe = re.findall('torrentid=([0-9]+)">The exact same torrent file already exists on the site!</a>$', SMerrorTorrent[0])
            if dupe:
                raise Exception(f'The exact same torrent file already exists on the site! See: https://sugoimusic.me/torrents.php?torrentid={dupe[0]}')
            else:
                raise Exception(SMerrorTorrent[0])
```

What needs to be done is to use beautifulsoup against `https://sugoimusic.me/torrents.php?torrentid={dupe[0]}` , and download the existing torrent and place it in `output.file_dir['smtorrents']` 

There is similar code that downloads the JPS torrent:

```
        torrentlink = html.unescape(gettorrentlink(torrentid))
        torrentfile = s.retrieveContent("https://jpopsuki.eu/%s" % torrentlink)  # Download JPS torrent
        torrentfilename = get_valid_filename(
            "JPS %s - %s - %s.torrent" % (torrentgroupdata.artist, torrentgroupdata.title, "-".join(releasedata)))
        torrentpath = Path(output.file_dir['jpstorrents'], torrentfilename)
```

Ideally SM and JPS torrent DLing should be unified into a single def, but this might too big a job right now.

When they are not dupes SM torrents are normally downloaded in `downloaduploadedtorrents()` , this handles multiple torrents, when users upload multiple torrents in a single group.

**Comment by damonjavert** [comment]
efb1562547d822bd80af827fb14696d0db5e2c95
04d875d070ced9ae638b8098ac1f42d213127618
